auraed/src/cells/cell_service/cell_service.rs

```diff
@@ -60,22 +60,30 @@ use tokio::sync::Mutex;
 use tonic::{Code, Request, Response, Status};
 use tracing::{info, trace, warn};
 
+/**
+ * Macro to perform an operation within a cell.
+ * It retries the operation with an exponential backoff strategy in case of connection errors.
+ */
 macro_rules! do_in_cell {
     ($self:ident, $cell_name:ident, $function:ident, $request:ident) => {{
+        // Lock the cells mutex to ensure exclusive access
```

Comment:
in general I'm not a huge fan of comments that describe the code that is already readable.  I prefer to keep comments for methods/classes and any tricky bits of code.  that way readers know to pay attention to the comments that exist and they don't become background noise that readers ignore.  does that make sense?